Fixed issue where custom resources requesting -1/-2 didn't work

bhroam opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
The scheduler uses special values for unspecified and infinity values (-1/-2).  The values they used are actually valid values that can be requested by a job for a custom resource.   The default value for a resource is infinity (-2).  If a job requested -2, the scheduler would set the value to the resource. It would then think it failed to set it since it was still set to the default value.  The job would be considered invalid for the cycle and be ignored.

#### Describe Your Change
The values for UNSPECIFIED and SCHD_INFINITY are too widely used on too many data types to change them to some sort of hugely negative or positive value.  It would require changing everything from a 5 bit field to many ints to longs.  The resource data type is a double.  There is a special floating point value called HUGE_VAL.  This is usually used for errors from floating point math functions.  I used HUGE_VAL for SCHD_INFINITY_RES and -HUGE_VAL for UNSPECIFIED_RES.

The customer who reported this bug used the custom resources in the job_sort_formula, so I wrote a generic job_sort_formula having a job request -2.  The test would fail if the job was ignored since the order of the two jobs would not be correct.

#### Attach Test and Valgrind Logs/Output
[job_sort_formula.log](https://github.com/PBSPro/pbspro/files/3219014/job_sort_formula.log)

agrawalravi90 commented on test/tests/functional/pbs_job_sort_formula.py:
```diff
@@ -0,0 +1,71 @@
+    def test_job_sort_formula_negative_value(self):
+        """
+        Test to see that negative values in the job_sort_formula
+        correctly sort properly
```
"correctly sort properly" sounds a bit excessive, i think you should remove either correctly or properly.

agrawalravi90 commented on test/tests/functional/pbs_job_sort_formula.py:
```diff
@@ -0,0 +1,71 @@
+
+        j1 = Job(TEST_USER, attrs={'Resource_List.foo': -2})
+        jid1 = self.server.submit(j1)
+        j2 = Job(TEST_USER, attrs={'Resource_List.foo': 1})
```
-1 right?

bhroam replied:
I meant 1 to be higher than -2, but I see your point.  I'll change it to -1.  The issue where the job was being ignored really had to do with -2, but testing them both is better.

bhroam commented on src/scheduler/constant.h:
```diff
@@ -92,13 +94,22 @@ enum select_job_status
+#define SCHD_INFINITY_RES HUGE_VAL
 #define SCHD_INFINITY_STR "SCHD_INFINITY"
 
+#define UNSPECIFIED -1
```
I don't think so.  They'd be defined to the same thing.  I think until we need to have them be separate strings, we should just use the macros we have now.

bhroam commented:
@agrawalravi90 I addressed your comments.  Please re-review.
[job_sort_formula.log](https://github.com/PBSPro/pbspro/files/3230137/job_sort_formula.log)
